hyperwallet/utils/apiclient.py

```diff
@@ -113,6 +113,8 @@ def _makeRequest(self,
         if response.status_code is 204:
             return {}
 
+        self.__checkResponseHeaderContentType(response)
```

Comment:
check the `headers=None` looks like its not set and made python decide on it, can we control this and use the appropriate `Accept` and `Content-type` headers based on encrypted or un encrypted request?

Reply:
`_makeRequest` is reference but header information is not explicitly set? and might be using auto detection feature for whatever payload we push in our http requerst from python library, I strongly suggest controlling this ourselves so that we will have a consistent request/response header information.

Reply:
In makeRequest method headers param is the additional headers, so it can be None. The base headers which are set for each request by default are stored in self.baseHeaders (line 42 in apiclient.py). As you propose 'Accept' header is set to 'application/jose+json' or 'application/json' depending on encryption enabled or not ( see self.baseHeaders )